src/mqtt/mqtt_test.c

```diff
@@ -982,7 +982,7 @@ TEST( MqttTest, MQTT_Subscribe_Publish_With_Qos_1 )
             /* Timeout. */
             break;
         }
-        else if( ( receivedPubAck != 0 ) && ( strncmp( TEST_MQTT_TOPIC, incomingInfo.pTopicName, TEST_MQTT_TOPIC_LENGTH ) == 0 ) )
+        else if( ( receivedPubAck != 0 ) && ( incomingInfo.topicNameLength == TEST_MQTT_TOPIC_LENGTH ) )
```

Comment:
This condition is to check if incomming publish is received. The content is checked outside while loop. I would suggest to use the following and comment the content is verified outside of the while loop. Minimun MQTT topic length is 1.
```suggestion
        else if( ( receivedPubAck != 0 ) && ( incomingInfo.topicNameLength > 0 ) )
```